crate fails to build on 32-bit architectures

For context: I'm trying to build deno, and by extension, rusty_v8, to eventually provide them as RPM packages for Fedora Linux.

I'm running `cargo build` with `V8_FROM_SOURCE=1` and `CLANG_BASE_PATH=/usr` so it uses the clang, gn, and ninja binaries provided by the system (clang 12.0.0-rc5).

I've now encountered a build issue in the rusty_v8 crate, in the bundled v8 code, particular to 32-bit architectures (32-bit x86 / i686 and 32-bit arm / armv7hl), possibly caused by `unsigned int` not having the expected size on those architectures:

```
  ../../../src/binding.cc:55:1: error: static_assert failed due to requirement 'sizeof(v8::Location) == sizeof(unsigned int) * 1' "Location size mismatch"
```

The complete build output for the object that fails to compile:

```
  [681/1785] CXX obj/rusty_v8/binding.o
  FAILED: obj/rusty_v8/binding.o 
  ../../../../../../../usr/bin/clang++ -MMD -MF obj/rusty_v8/binding.o.d -DUSE_UDEV -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_OZONE=1 -DUSE_X11=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -D_LIBCPP_ABI_UNSTABLE -D_LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS -D_LIBCXXABI_DISABLE_VISIBILITY_ANNOTATIONS -D_LIBCPP_ENABLE_NODISCARD -D_LIBCPP_HAS_NO_VENDOR_AVAILABILITY_ANNOTATIONS -DCR_LIBCXX_REVISION=8fa87946779682841e21e2da977eccfb6cb3bded -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DV8_DEPRECATION_WARNINGS -DV8_TARGET_ARCH_ARM -DCAN_USE_ARMV7_INSTRUCTIONS -DCAN_USE_VFP3_INSTRUCTIONS -DCAN_USE_VFP32DREGS -DCAN_USE_NEON -DV8_HAVE_TARGET_OS -DV8_TARGET_OS_LINUX -DDISABLE_UNTRUSTED_CODE_MITIGATIONS -DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=64 -DENABLE_MINOR_MC -DV8_INTL_SUPPORT -DV8_ATOMIC_OBJECT_FIELD_WRITES -DV8_ATOMIC_MARKING_STATE -DV8_ENABLE_LAZY_SOURCE_POSITIONS -DV8_SHARED_RO_HEAP -DV8_WIN64_UNWINDING_INFO -DV8_ENABLE_REGEXP_INTERPRETER_THREADED_DISPATCH -DV8_SNAPSHOT_COMPRESSION -DV8_ENABLE_SYSTEM_INSTRUMENTATION -DV8_ENABLE_WEBASSEMBLY -DV8_DEPRECATION_WARNINGS -DV8_DEPRECATION_WARNINGS -DV8_DEPRECATION_WARNINGS -DV8_DEPRECATION_WARNINGS -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DUSE_CHROMIUM_ICU=1 -DU_ENABLE_TRACING=1 -DU_ENABLE_RESOURCE_TRACING=0 -DU_STATIC_IMPLEMENTATION -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -DUCHAR_TYPE=uint16_t -I../../.. -Igen -I../../../v8 -Igen/v8 -I../../../v8/include -Igen/v8/include -I../../../v8/include -Igen/v8/include -Igen/v8/include -I../../../v8/include -I../../../v8/include -Igen/v8/include -I../../../third_party/icu/source/common -I../../../third_party/icu/source/i18n -fno-delete-null-pointer-checks -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -funwind-tables -fPIC -pthread -fcolor-diagnostics -fmerge-all-constants -fcrash-diagnostics-dir=../../../tools/clang/crashreports -mllvm -instcombine-lower-dbg-declare=0 -Xclang -fno-experimental-new-pass-manager -fcomplete-member-pointers --target=arm-linux-gnueabihf -march=armv7-a -mfloat-abi=hard -mtune=generic-armv7-a -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -Xclang -fdebug-compilation-dir -Xclang . -no-canonical-prefixes -mfpu=neon -mthumb -Wall -Wextra -Wimplicit-fallthrough -Wunreachable-code -Wthread-safety -Wextra-semi -Wno-missing-field-initializers -Wno-unused-parameter -Wno-c++11-narrowing -Wno-unneeded-internal-declaration -Wno-undefined-var-template -Wno-psabi -Wno-ignored-pragma-optimize -Wno-implicit-int-float-conversion -Wno-final-dtor-non-final-class -Wno-builtin-assume-aligned-alignment -Wno-deprecated-copy -Wno-non-c-typedef-for-linkage -Wno-max-tokens -O2 -fno-ident -fdata-sections -ffunction-sections -fno-omit-frame-pointer -g1 -gno-inline-line-tables -fvisibility=hidden -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -fcolor-diagnostics -fansi-escape-codes -Wmissing-field-initializers -Wunreachable-code -std=c++14 -fno-trigraphs -Wno-trigraphs -fno-exceptions -fno-rtti -nostdinc++ -isystem../../../buildtools/third_party/libc++/trunk/include -isystem../../../buildtools/third_party/libc++abi/trunk/include -fvisibility-inlines-hidden -c ../../../src/binding.cc -o obj/rusty_v8/binding.o
  ../../../src/binding.cc:55:1: error: static_assert failed due to requirement 'sizeof(v8::Location) == sizeof(unsigned int) * 1' "Location size mismatch"
  static_assert(sizeof(v8::Location) == sizeof(size_t) * 1,
  ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  1 error generated.
```

The NodeJS package on Fedora also contains a bundled V8 library (version 8.4.371.19), which does not exhibit this problem.

System information:

- Fedora Linux 35 / Rawhide
- compiles fine on x86_64 (64-bit x86) and aarch64 (64-bit arm)
- fails to compile on i686 (32-bit x86) and armv7hl (32-bit arm)

Relevant versions:

- clang, llvm, lld 12.0.0-rc5
- Rust 1.51.0
- ninja 1.10.2
- glib2 2.68.1
- python 3.9

--- a/src/binding.cc
+++ b/src/binding.cc
@@ -52,7 +52,7 @@ static_assert(sizeof(v8::ReturnValue<v8::Value>) == sizeof(size_t) * 1,
 static_assert(sizeof(v8::TryCatch) == sizeof(size_t) * 6,
               "TryCatch size mismatch");
 
-static_assert(sizeof(v8::Location) == sizeof(size_t) * 1,
+static_assert(sizeof(v8::Location) == sizeof(int) * 2,
               "Location size mismatch");
 
 static_assert(sizeof(v8::SnapshotCreator) == sizeof(size_t) * 1,

--- a/src/module.rs
+++ b/src/module.rs
@@ -195,7 +195,7 @@ extern "C" {
 /// A location in JavaScript source.
 #[repr(C)]
 #[derive(Debug)]
-pub struct Location([usize; 1]);
+pub struct Location([i32; 2]);
 
 impl Location {
   pub fn get_line_number(&self) -> int {

